main.tf

```diff
@@ -1,13 +1,92 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+provider "awsutils" {
+  region = "us-east-1"
+}
+
+module "tls_self_signed" {
+  source = "./modules/tls-self-signed"
+
+  organization_name = var.organization_name
+}
+
+resource "aws_acm_certificate" "ca" {
+  private_key      = module.tls_self_signed.ca_private_key_pem
+  certificate_body = module.tls_self_signed.ca_cert_pem
+}
+
+resource "aws_acm_certificate" "root" {
+  private_key       = module.tls_self_signed.root_private_key_pem
+  certificate_body  = module.tls_self_signed.root_cert_pem
+  certificate_chain = module.tls_self_signed.ca_cert_pem
+}
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+resource "aws_acm_certificate" "server" {
+  private_key       = module.tls_self_signed.server_private_key_pem
+  certificate_body  = module.tls_self_signed.server_cert_pem
+  certificate_chain = module.tls_self_signed.ca_cert_pem
+}
+
+resource "aws_ec2_client_vpn_endpoint" "default" {
+  description            = module.this.id
+  server_certificate_arn = aws_acm_certificate.server.arn
+  client_cidr_block      = var.client_cidr
+
+  authentication_options {
+    type                       = "certificate-authentication"
+    root_certificate_chain_arn = aws_acm_certificate.root.arn
   }
+
+  connection_log_options {
+    enabled               = var.logging_enabled
+    cloudwatch_log_group  = var.logging_enabled ? aws_cloudwatch_log_group.vpn.name : null
+    cloudwatch_log_stream = var.logging_enabled ? aws_cloudwatch_log_stream.vpn.name : null
+  }
+
+  tags = module.this.tags
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+resource "aws_ec2_client_vpn_network_association" "default" {
+  client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.default.id
+  subnet_id              = var.aws_subnet_id
+}
+
+resource "aws_ec2_client_vpn_authorization_rule" "internal" {
+  client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.default.id
+  target_network_cidr    = var.aws_authorization_rule_target_cidr
+  authorize_all_groups   = true
+}
+
+resource "aws_ec2_client_vpn_authorization_rule" "internet_rule" {
+  count = var.internet_access_enabled ? 1 : 0
+
+  client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.default.id
+  target_network_cidr    = "0.0.0.0/0"
+  authorize_all_groups   = true
+}
+
+resource "aws_ec2_client_vpn_route" "internet_route" {
+  count = var.internet_access_enabled ? 1 : 0
+
+  client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.default.id
+  destination_cidr_block = "0.0.0.0/0"
+  target_vpc_subnet_id   = aws_ec2_client_vpn_network_association.default.subnet_id
+}
+
+resource "aws_cloudwatch_log_group" "vpn" {
+  name              = "/aws/vpn/${module.this.id}/logs"
+  retention_in_days = var.logs_retention
+  tags              = module.this.tags
+}
+
+resource "aws_cloudwatch_log_stream" "vpn" {
+  name           = "${module.this.id}-vpn-usage"
```

Comment:
we should use labels for this because even the `-` is a parameter of labels.

Reply:
you can add `vpn` and `usage` to the attributes

Reply:
https://github.com/cloudposse/terraform-null-label/blob/cf38625a5dde227db04c9cfedc1327d610229fec/examples/complete/label7.tf#L17

Reply:
I think `usage` doesn't add any extra disambiguation. It's already clear that it's a log events.

Reply:
Actually, lets use this module https://github.com/cloudposse/terraform-aws-cloudwatch-logs